pkg/apiserver/webhook/cloudentityselector_webhook.go

```diff
@@ -250,24 +218,31 @@ func (v *CESValidator) validateDelete(_ admission.Request) admission.Response {
 	return admission.Allowed("")
 }
 
-// GetOwnerAccount fetches the CPA account owning the current CES.
-func (v *CESValidator) GetOwnerAccount(selector *v1alpha1.CloudEntitySelector) (*v1alpha1.CloudProviderAccount, error) {
-	accountNameSpacedName := &types.NamespacedName{
-		Namespace: selector.Namespace,
+// getReferencedAccount fetches the CPA account referenced by the current CES.
+func (v *CESValidator) getReferencedAccount(selector *v1alpha1.CloudEntitySelector) (*v1alpha1.CloudProviderAccount, error) {
+	accountNamespacedName := &types.NamespacedName{
+		Namespace: selector.Spec.AccountNamespace,
 		Name:      selector.Spec.AccountName,
 	}
-	ownerAccount := &v1alpha1.CloudProviderAccount{}
-	err := v.Client.Get(context.TODO(), *accountNameSpacedName, ownerAccount)
+	referencedAccount := &v1alpha1.CloudProviderAccount{}
+	err := v.Client.Get(context.TODO(), *accountNamespacedName, referencedAccount)
 	if err != nil {
-		v.Log.Error(err, errorMsgOwnerAccountNotFound, "CloudEntitySelector", selector,
-			"account", *accountNameSpacedName)
+		v.Log.Error(err, errorMsgReferencedAccountNotFound, "CloudEntitySelector", selector,
+			"account", *accountNamespacedName)
 		return nil, err
 	}
-	return ownerAccount, err
+	return referencedAccount, err
 }
 
 // validateMatchSections checks for unsupported selector match combinations and errors out.
 func (v *CESValidator) validateMatchSections(selector *v1alpha1.CloudEntitySelector) error {
+	// Empty vpcMatch and empty vmMatch section are not supported.
+	for _, m := range selector.Spec.VMSelector {
+		if m.VpcMatch == nil && len(m.VMMatch) == 0 {
+			return fmt.Errorf("%s", errorMsgVpcOrVmMatchNotAvailable)
```

Comment:
Is the error returned here user facing - can you show how the error would look like?

Reply:
Error from server: error when creating "ces_azure2.yaml": admission webhook "vcloudentityselector.kb.io" denied the request: either vpcMatch or vmMatch is mandatory